tests/5.0/loop/test_loop_reduction_multiply.F90

```diff
@@ -0,0 +1,85 @@
+!===--- test_loop_reduction_multiply.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test uses the reduction clause on a loop directive, testing that the
+! variable in the reduction clause is properly reduced using the multiply
+! operator.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_loop_reduction_multiply
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_VERBOSE(test_multiply() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION test_multiply()
+    INTEGER,DIMENSION(N):: a, num_threads
+    INTEGER:: errors, device_result, host_result, x, y
+    CHARACTER(len=400) :: msgHelper
+    INTEGER :: seedSize
+    INTEGER,ALLOCATABLE :: seed(:)
+    DOUBLE PRECISION:: randomNumber
+
+    errors = 0
+    CALL random_seed(size=seedSize)
+    ALLOCATE(seed(seedSize))
+    seed = 1
+    CALL random_seed(put=seed)
+    DEALLOCATE(seed)
+
+    DO x = 1, N
+       !random_number() generates a real number, r, uniformly distributed in 0 <= r < 1.
+       CALL random_number(randomNumber)
+       IF ( randomNumber .eq. 0.0 ) THEN
+         a(x) = 2
+       ELSE
+         a(x) = 1
+       ENDIF
+       num_threads(x) = -1 * x
+    END DO
+
+    device_result = 1
+
+    DO x = 1, N, 16
+      device_result = 1
+      !$omp parallel num_threads(OMPVV_NUM_THREADS_HOST)
+      !$omp loop reduction(*:device_result)
+      DO y = 1, 16
```

Comment:
Works with GCC 12 after applying the `y` range fixes. With host fallback, the result is then:
```
[OMPVV_INFOMSG ompvv.F90:256] Test ran with no errors
[OMPVV_RESULT test_loop_reduction_multiply.F90] Test passed on the host.
```

```suggestion
      DO y = 0, 15
```

Reply:
@tob2 Thank you for the nice catch!